lib/proto/attach-form.js

```diff
@@ -44,23 +44,19 @@ var formAction = function(el, event) {
 
   event.preventDefault()
 
-  var paramObject = []
+  var params = []
   for (var elementKey in el.elements) {
     var element = el.elements[elementKey]
     // jscs:disable disallowImplicitTypeConversion
     if (!!element.name && element.attributes !== undefined && element.tagName.toLowerCase() !== "button") {
       // jscs:enable disallowImplicitTypeConversion
-      if ((element.attributes.type !== "checkbox" && element.attributes.type !== "radio") || element.checked) {
-        paramObject.push({name: encodeURIComponent(element.name), value: encodeURIComponent(element.value)})
+      if ((!element.attributes.type || element.attributes.type.value !== "checkbox" && element.attributes.type.value !== "radio") || element.checked) {
+        params.push({name: encodeURIComponent(element.name), value: encodeURIComponent(element.value)})
       }
     }
   }
 
-  // Creating a getString
-  var paramsString = (paramObject.map(function(value) {return value.name + "=" + value.value})).join("&")
-
-  options.requestOptions.requestPayload = paramObject
-  options.requestOptions.requestPayloadString = paramsString
+  options.requestOptions.requestParams = params
```

Comment:
@robinnorth Would this be considered a breaking change?

Reply:
Technically maybe, but as we don't document the `requestOptions` option or its keys, so the implication could be that you shouldn't rely on it to have a certain shape. I'm not sure that it's actually very useful to have `requestOptions` be a part of `this.options` anyway, as it's only intended to be set internally by the library. As #127 and #130 show, it just causes issues with manual invocation when it's not set. Perhaps it should be refactored to `this.requestOptions`?

Reply:
Do we need a global `requestOptions` at all? Why don't we just pass it directly to `loadUrl()`?

Reply:
No, I don't think we do need it to be global -- it can just be a third parameter to `loadUrl()`, after `options`, and default to an empty object if not set.

Reply:
That would be simpler.

I don't think we need to pass `options` either. When it's needed, it can be retrieved from `this.options`.

Reply:
Currently the history popstate handler relies on being able to pass a modified options object to `loadUrl`, so I don't think we can easily drop that parameter.

Reply:
Good point. But if it's not passed in, we can default to `this.options`.